### ISSUE TITLE
create_new_visit_when_website_referrer_changes = 1, but it doesn't seem to work.

### ISSUE THREAD

**Issue by jjp0226** [piwik/piwik#11605]
I have already modified config/config.ini.php (and global.ini.php) file as follows.
https://piwik.org/faq/how-to/faq_19616/

I expected to record as a new visit if the visitor comes back to the site less than 30 min ago with the same browser but a different campaign parameters.

But it doesn't seem to work.

Please, help me.

**Comment by sgiehl** [comment]
Which Piwik version do you use and are you using any additional plugin for campaign tracking?

**Comment by JamesJungmoonLim** [comment]
@sgiehl I also have the same problem, I tested community version and cloud version(trial) both.

**Comment by jjp0226** [comment]
@sgiehl I'm testing on Piwik 3.0.3 with Marketing Campaigns Reporting Plug-in.

**Comment by sgiehl** [comment]
@jjp0226  setting `create_new_visit_when_website_referrer_changes` to 1 won't have any effect on campaigns. This effects all referrers without campaign data.

But with the default settings of Piwik (`create_new_visit_when_campaign_changes = 1`) a new visit should be forced when the campaign data changes. We need to investigate why this might not be the case.

@JamesJungmoonLim @jjp0226 could you post the url parameters you tried? Couldn't reproduce that locally.

**Comment by JamesJungmoonLim** [comment]
@sgiehl    
[(https://s3.ap-northeast-2.amazonaws.com/jmlim-echo-test/test.html?pk_campaign=test1&pk_kwd=test1)]
[(https://s3.ap-northeast-2.amazonaws.com/jmlim-echo-test/test.html?pk_campaign=test2&pk_kwd=test2)]
these are the urls that I tried

thx

**Comment by jjp0226** [comment]
@sgiehl I tried as follows:

Test Page: http://cresendo.net/test.html
URL Parameters: http://cresendo.net/test.html?pk_campaign=campaign1&pk_kwd=keyword1

I have modified config/config.ini.php (and global.ini.php) file by create_new_visit_when_campaign_changes=1

**Comment by JamesJungmoonLim** [comment]
@sgiehl  Is there any update on this?

**Comment by sgiehl** [comment]
Sorry, hadn't had time yet to have a deeper look at it.
Did any of you change/overwrite the campaign configuration in the config file?
Looking at the code a new visit should be forced with your given urls and default campaign parameters.

**Comment by feelydh** [comment]
@sgiehl Like jjp0226 and JamesJungmoonLim, I have the same problem

So I tested it,
(Create_new_visit_when_campaign_changes = 1) seems to work after 30 minutes of default session time, right?

I eventually changed the piwik.js source code myself

Looking at the code, we do not store the values ​​of the pk_campaign and pk_key parameters in the cookie when there is a session cookie (ses)

Regardless of the session cookie, we changed the cookie back to save if the value of the pk_campaign parameter differs from the value stored in the cookie

example
" if (!cookieSessionValue){ " ==>" if (!cookieSessionValue || {The corresponding conditional statement}){ "

After modifying, it was tested again,
Regardless of the default session time of 30 minutes Create a new visit every time a user visits website with a new campaign.

(Create_new_visit_when_campaign_changes = 1) Is the purpose of this setting the same?
Or is there a way to get the same result without doing it like me?
